sdarq/backend/src/dojo_helper.py

```diff
@@ -0,0 +1,37 @@
+import json
+import logging
+import os
+
+import requests
+
+dojo_host = os.getenv('dojo_host')
+dojo_api_key = os.getenv('dojo_api_key')
+headers = {
+    "content-type": "application/json",
+    "Authorization": f"Token {dojo_api_key}",
+}
+
+
+products_endpoint = f"{dojo_host}api/v2/products/"
+
+def dojo_create_or_update(name, description, product_type, user_email):
+    data = {
+        'name': name,
+        'description': description,
+        'prod_type': product_type}
+    response = requests.get(products_endpoint+"?name_exact="+name, headers=headers)
+    if response.json()['count'] > 0:
+        id = response.json()['results'][0]['id']
+        res = requests.patch(products_endpoint+"/"+str(id), data=json.dumps(data))
```

Comment:
Looks like it does not fail if users try to create a new service with the same name, but it does not update the description with the new data.

Reply:
It was missing the headers, thanks for catching this. 
It's still not working, but the reasons are more mysterious now.

Reply:
It works now! https://defectdojo.dsp-appsec-dev.broadinstitute.org/product/323